src/vim9class.c

```diff
@@ -2583,6 +2583,63 @@ member_lookup(
 	return object_member_lookup(cl, name, namelen, idx);
 }
 
+/*
+ * Find the class that defines the named member. Look up the hierarchy
+ * starting at "cl".
+ *
+ * Return the class that defines the member "name", else NULL.
+ * Fill in "p_typ" if specified, for found class.
+ * "p_typ" is either VAR_OBJECT or VAR_CLASS; VAR_UNKNOWN if not found
+ */
+
+// TODO: if requested could indirectly return ocmember_T and idx.
+
+    static class_T *
+class_defining_member(class_T *cl, char_u *name, size_t len,
+					    vartype_T *p_typ, ocmember_T **p_m)
```

Comment:
It looks like `p_typ` is always NULL? Is this parameter needed?

Reply:
Not really; but it's the only way for a caller to determine if the name is object or class. The function is static, so no reason to provide more features than needed.

Shall I take it out and leave a not about how it can be extended?

BTW, I was just looking and I think I can cover 2632. It's a path that should be tested. Shouldn't need to change any code, just the test. (famous last words)